packages/migrate/src/__tests__/DbPull.test.ts

```diff
@@ -481,6 +481,231 @@ describe('postgresql - missing database', () => {
+      expect(workWorkersView).toMatchInlineSnapshot(`
+        SELECT
+          su.first_name,
+          su.last_name,
+          c.name AS company_name
+        FROM
+          (
+            someuser su
+            LEFT JOIN WORK.company c ON ((su.company_id = c.id))
+          );
+      `)
+
+      expect(ctx.mocked['console.log'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+      expect(ctx.mocked['console.info'].mock.calls.join('\n')).toMatchInlineSnapshot(`
+        Prisma schema loaded from schema.prisma
+        Datasource "db": PostgreSQL database "tests-migrate-db-pull", schemas "public, work" at "localhost:5432"
+      `)
+      expect(ctx.mocked['console.error'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+      expect(ctx.mocked['process.stdout.write'].mock.calls.join('\n')).toMatchInlineSnapshot(`
+
+
+                        - Introspecting based on datasource defined in schema.prisma
+
+                        ✔ Introspected 2 models and wrote them into schema.prisma in XXXms
+                              
+                        *** WARNING ***
+
+                        The following views were ignored as they do not have a valid unique identifier or id. This is currently not supported by the Prisma Client. Please refer to the documentation on defining unique identifiers in views: https://pris.ly/d/view-identifiers
+                        - View "simpleuser"
+                        - View "workers"
+
+                        Run prisma generate to generate Prisma Client.
+
+                  `)
+      expect(ctx.mocked['process.stderr.write'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+    })
+
+    test('extraneous files in views folder should be removed on introspect', async () => {
+      ctx.fixture(path.join(fixturePath))
+
+      await ctx.fs.dirAsync('views')
+      const initialList = await ctx.fs.listAsync('views')
+      expect(initialList).toMatchInlineSnapshot(`[]`)
+
+      await ctx.fs.dirAsync('views/extraneous-dir')
+      await ctx.fs.fileAsync('views/extraneous-file.sql')
+      const extraneousList = await ctx.fs.listAsync('views')
+      expect(extraneousList).toMatchInlineSnapshot(`
+        [
+          extraneous-dir,
+          extraneous-file.sql,
+        ]
+      `)
+
+      const introspect = new DbPull()
+      const result = introspect.parse([])
+      await expect(result).resolves.toMatchInlineSnapshot(``)
+
+      // the folders in `views` match the database schema names (public, work) of the views
+      // defined in the `setup.sql` file
+      const list = await ctx.fs.listAsync('views')
+      expect(list).toMatchInlineSnapshot(`
+        [
+          public,
+          work,
+        ]
+      `)
+
+      const tree = await ctx.fs.findAsync({ directories: false, files: true, recursive: true, matching: 'views/**/*' })
+      expect(tree).toMatchInlineSnapshot(`
+        [
+          views/public/simpleuser.sql,
+          views/work/workers.sql,
+        ]
```

Comment:
These two fails on Windows. How can we not run this test on Windows or skips these snapshots?

Reply:
I don't think we should skip them since it's an actual bug on windows. The reason is probably because of the broken glob pattern on windows (see the review comment below).